src/main/java/com/google/sps/servlets/utils/CommentAnalysis.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets.utils;
+
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This class encapsulates each element in json comment array into separate user comment object
+ * and does sentiment analysis on each of them.
+ */
+public class CommentAnalysis {
+  private CommentThreadListResponse youtubeResponse;
+  private LanguageServiceClient languageService;
+
+  public CommentAnalysis() throws IOException {
+    this.languageService = LanguageServiceClient.create();
+  }
+
+  /**
+   * It computes an overall statistics score from the retrieved youtube comments.
+   * @return a Statistics object that contains required values to display
+   */
+  public Statistics computeOverallStats(CommentThreadListResponse youtubeResponse) {
+    ArrayList<Double> scoreValues = new ArrayList<>();
+    for (CommentThread commentThread: youtubeResponse.getItems()) {
+      UserComment userComment = new UserComment(commentThread);
+      try {
+        scoreValues.add(calculateSentiAnalysisScore(userComment));
+      } catch (IOException e) {
+        System.err.println(e.getLocalizedMessage());
+      }
```

Comment:
Generally a good engineering principle to follow is to "fail-fast" (https://en.wikipedia.org/wiki/Fail-fast). 

This means that if any kind of error is encountered, exceptions should generally just be propagated up to the top by individual components (e.g. objects, services). Generally this should be the case unless there are some classes of exceptions that are deemed to be recoverable.

For this specific example, we shouldn't swallow the thrown `IOException` since it will hide the fact that a call to the LanguageService API failed to consumers of the `computeOverallStats` method. If we do want to have some exception handling logic (for non-recoverable errors), it makes sense to put it at the highest level possible (i.e. in YoutubeServlet).

Reply:
Updated: directly throws the IOException at the top of funciton defintion